cfme/infrastructure/virtual_machines.py

```diff
@@ -997,17 +1099,19 @@ def am_i_here(self, *args, **kwargs):
             provider = self.obj.provider.name
         else:
             raise DestinationNotFound("the destination isn't found")
-        return match_page(summary='VM or Templates under Provider "{}"'.format(provider))
+        return (self.view.sidebar.vmstemplates.is_opened and self.view.entities.title.text ==
```

Comment:
I think this check needs to move into `view.is_displayed`, and you should call it directly here as `self.view.is_displayed`, so that `view.is_displayed` is accurate when called outside of navigation context as well.